### PR TITLE
Rendering2

### PR DESCRIPTION
**Pull request by hamishs**
Refactors the OpenCV viewer so that it is much faster (almost 8x faster) and a bit easier to customise.

Speed improvements:
* Cache the global coordinates of each road network geometry within each scenario so drawing each frame only requires translating the coords to the local frame.
* Remove the need to make the frame contiguous before adding text by removing `.flip` and making `vec2pix` use the correct x-axis orientation.

Changes to the default rendering behaviour:
* Black background.
* Lane lines and road boundary rendered.
* Buildings rendered.
* The front of each entity is indicated with a block (colour controlled by `entity_front_color`).

### CODE REVIEW

File: scenario_gym/viewer/opencv.py [hunk 1]
```diff
@@ -191,109 +181,229 @@ def render(self, state: State) -> Optional[int]:
...
+        xy_front = vec2pix(front_bbox, mag=self.mag, h=self.h, w=self.w)
+        cv2.fillPoly(self._frame, [xy], c)
+        cv2.fillPoly(self._frame, [xy_front], self.entity_front_color)
+
+    def render_text(self, state: State) -> None:
+        """Add text to the frame."""
+        v = np.linalg.norm(state.scenario.entities[0].velocity[:3])
+        cv2.putText(
+            self._frame,
+            "Ego speed: {:.4f}".format(v),
```
Comment (LNS98) [inline scenario_gym/viewer/opencv.py:282]: 2?

File: tests/test_scenario_gym.py [hunk 2]
```diff
@@ -44,10 +44,23 @@ def test_render(all_scenarios):
 
     gym = ScenarioGym(timestep=0.1)
     gym.load_scenario(scenario_path)
-    gym.reset_scenario()
     gym.rollout(render=True)
+    gym.rollout(
+        render=True,
+        video_path=scenario_path.replace("Scenarios", "Recordings").replace(
+            ".xosc", "_2.mp4"
```
Comment (LNS98) [inline tests/test_scenario_gym.py:51]: why the _2?

**Review by LNS98 — COMMENTED** [review]
I have had a quick skim through it - and yeah looks a lot better, have very minor comments:

File: scenario_gym/viewer/opencv.py (hunk 1 above)
Comment (hamishs) [reply scenario_gym/viewer/opencv.py:282]: sorry where does the 2 fit in here? the rounding?

File: tests/test_scenario_gym.py (hunk 2 above)
Comment (hamishs) [reply tests/test_scenario_gym.py:51]: Thank you this was me debugging will fix.

**Pull request merged**